### PR TITLE
Add unit tests for FileStorage

### PR DESCRIPTION
**Pull request by igorkravchenko**
This PR adds unit tests for FileStorage. It also moves all mocks under `#if DEBUG` to avoid any of them to slip in into live implementation. In addition it adds `FoundationBased` namespace for dependencies like FileManager and Data. The idea to use it whenever there is a need to mock Foundation dependencies using structs and closures.

MOB-1128

### CODE REVIEW

File: GliaWidgets/CoreSDKClient/CoreSDKClient.Mock.swift [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import Foundation
```
Comment (dukhovnyi) [inline GliaWidgets/CoreSDKClient/CoreSDKClient.Mock.swift:2]: let's make a newline after the import section.

File: GliaWidgets/FoundationBased.Mock.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#if DEBUG
```
Comment (dukhovnyi) [inline GliaWidgets/FoundationBased.Mock.swift:1]: would you like to avoid having this code in the release schema?

**Review by dukhovnyi — APPROVED** [review]
Looks good for me

File: GliaWidgetsTests/Lib/Storage/FileSystemStorageTests.swift [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        fileStorage.environment.fileManager.copyItemAtPath = { _, toPath in
+            copiedPathResult = toPath
+        }
+        fileStorage.store(from: expectedFileUrl, for: expectedKey)
+
+        XCTAssertEqual(copiedPathResult, expectedFileUrl.path)
+        XCTAssertEqual(removedUrlResult, expectedFileUrl)
+    }
+
+    func test_storeFromUrlForKeyCreatesIntermediateDirsForFilePathIfNeeded () {
```
Comment (dukhovnyi) [inline GliaWidgetsTests/Lib/Storage/FileSystemStorageTests.swift:232]:
```suggestion
    func test_storeFromUrlForKeyCreatesIntermediateDirsForFilePathIfNeeded() {
```

File: GliaWidgets/CoreSDKClient/CoreSDKClient.Mock.swift (hunk 1 above)
Comment (igorkravchenko) [reply GliaWidgets/CoreSDKClient/CoreSDKClient.Mock.swift:2]: Fixed 7c23e635f77daa3dacb46bcaaab2e852d2ed04e8

File: GliaWidgetsTests/Lib/Storage/FileSystemStorageTests.swift (hunk 3 above)
Comment (igorkravchenko) [reply GliaWidgetsTests/Lib/Storage/FileSystemStorageTests.swift]: Fixed ea054ed4598879273b35ccef2742cae4c1b61323

**Pull request merged**